### ISSUE TITLE
generic tagging framework

### ISSUE THREAD

**Issue by anovis** [capitalone/Particle-Cloud-Framework#115]
seems like a lot of particles have a separate tagging function which causes extra logic for update and is_state_definition_equivalent. We should try to break that logic out.

**Comment by ethanwlo** [comment]
I agree... Rumor has it that AWS is working on a generic tagging resource that can be associated to any resource. We MIGHT want to wait for that and simply break out all tag into its own particle.

**Comment by kapilt** [comment]
the closest thing to a generic tagging service today  is called resource group tagging api, its been out for a few years.

**Comment by ethanwlo** [comment]
Yea. The problem with generic tagging service is that currently it only supports resources that already support tags. The “rumor” I heard is that it’s being expanded to support all resource types in the near future.
